Clarify catchless try in explainer (#168)
WebAssembly/exception-handling: proposals/exception-handling/Exceptions.md

`catch_all` block. All `catch` blocks must precede the `catch_all` block, if
any. The `catch`/`catch_all` instructions (within the try construct) are called
the _catching_ instructions. There may not be any `catch` or `catch_all` blocks

> after a `try`, in which case the `try` block does not catch any exceptions and
> rethrows them.

Reviewer: ```suggestion
after a `try`, in which case the `try` block does not catch any exceptions.
```
(I think the rethrow part is a bit off, since technically, a non-catch isn't the same as a rethrow.)

can also be written as a label, like branches. So `rethrow 0` in the example
above can also be written as `rethrow $l3`.

Note that `rethrow 2` is not allowed because it does not refer to a `try` label

> with a catch block. Rather, it references a `block` instruction, so it will

Reviewer: ```suggestion
from within its catch block. Rather, it references a `block` instruction, so it will
```

The `rethrow` here references `try $l2`, but the `rethrow` is not within its
`catch` block.

> Also, targetting a label of `catch`-less `try` or `try`-`delegate` is also a

Reviewer: I still don't see the relevance of the "catch-less" here, I think it's misleading: the example is a validation failure regardless of whether you add a catch clause or not. Both examples are invalid simply because rethrow does not reference a catch-block, no other reason.

Author: I guess the whole added paragraph here about `rethrow` is not really meaningful and also misleading as you suggested. This PR's goal was to clarify about catchless try and its relationship w/ `delegate`, so I'll delete this whole paragraph and example.

> validation failure, because `rethrow` is not within a `catch` or `catch_all`.

Reviewer: This part is confusing as well: even if there was a surrounding catch somewhere it would still be invalid. Whether a catch _exists_ is irrelevant, it only matters whether the rethrow _references_ a catch.

Author: Deleted.

Note that the example below is a validation failure:
```
try $l1
catch
  try
    call $foo
  delegate $l1  ;; (= delegate 0)
catch_all
  ...
end
```
Here `delegate` is trying to delegate to `catch`, which exists before the
`delegate`. The `delegate` instruction can only target `try` label whose
catching instructions (`catch`/`catch_all`/`delegate`) come after the
`delegate` instruction.

`delegate` can also target `catch`-less `try`, in which case the effect is the

> same as the `try` has catches but none of the catches are able to handle the

Reviewer: ```suggestion
same as if the `try` has catches but none of the catches are able to handle the
```
